config/routes.rb

```diff
@@ -5,13 +5,18 @@
     get '/file/auth/:id/:file_name' => 'webauth#login_file', as: :auth_file
   end
 
-  constraints file_name: %r{[^/][\w-]+} do
+  # known issue:  this will fail to match for file names with multiple dots, though
+  # the rails magic that populates `format` with the extension after the dot will work
+  # for files with one dot.  e.g. "/media/oo000oo0000/video.mp4" will work and have a
+  # `format` of "mp4", but "/media/oo000oo0000/adjective.video.mp4" will result in
+  # "No route matches".
+  constraints file_name: %r{[^/][\w\(\)\{\}\[\]\*\+\|\^ @:#%;=&$!,~"'_-]+} do
```

Comment:
if the test won't pass with square brackets, should we take them out of this?   

But really: why would this be different when the code is so similar to file#show ?   Maybe we can pair and poke around together?